lib/types.ts

```diff
@@ -470,11 +470,7 @@ export type XOR<T, U> = T | U extends object ? (Without<T, U> & U) | (Without<U,
 
 /** Functional programming essentials */
 export type Head<T extends AnyArray> = T["length"] extends 0 ? never : T[0];
-export type Tail<T extends AnyArray> = T["length"] extends 0
-  ? never
-  : ((...t: T) => void) extends (first: any, ...rest: infer Rest) => void
-  ? Rest
-  : never;
+export type Tail<T extends AnyArray> = T["length"] extends 0 ? never : T extends [any, ...infer Rest] ? Rest : never;
```

Comment:
You can even remove first check as it doesn't match next conditional type and will end up being `never` anyway

```suggestion
export type Tail<T extends AnyArray> = T extends [any, ...infer Rest] ? Rest : never;
```

Reply:
Aha! Good catch! Thanks.